### ISSUE TITLE
Error compiling in Mac OSX 10.5.7 PPC (endian.h)

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#626]
**Reported by misterx on 25 May 2009 01:39 UTC**

```
# make sqlite=true
(...)
gcc -g -W -Wall -Werror -O3 -D_FILE_OFFSET_BITS=64 -D_REVISION=-1  -I/usr/local/include -DHAVE_SQLITE -Iinclude   -c -o airdecloak-ng.o airdecloak-ng.c
In file included from airdecloak-ng.c:44:
include/radiotap-parser.h:20:20: error: endian.h: No such file or directory
make[*** [airdecloak-ng.o](1]:) Error 1
make: *** [all] Error 2

```

**Comment by aircrack-ng** [comment]
**Comment by anonymous on 25 May 2009 01:39 UTC**

gcc -g -W -Wall -Werror -O3 -D_FILE_OFFSET_BITS=64 -D_REVISION=1552  -I/usr/local/include -DHAVE_SQLITE -Iinclude   -c -o airdecloak-ng.o airdecloak-ng.c
In file included from airdecloak-ng.c:44:
include/radiotap-parser.h:20:20: error: endian.h: No such file or directory
make: *** [airdecloak-ng.o] Error 1

intel also

**Comment by aircrack-ng** [comment]
**Comment by anonymous on 25 May 2009 01:39 UTC**

Under MacOS X is should be #include <machine/endian.h> instead of #include <endian.h> in radiotap-parser.h.
This change fixes the compile issue.  May need to add a MacOS X check in the code to fix it.

**Comment by aircrack-ng** [comment]
**Comment by misterx on 25 May 2009 01:39 UTC**

Should be fixed by aircrack-ng/aircrack-ng@9085c8d721f0b0eb1c439992847369a8fbb8ce90
